osf/models/notification_type.py

```diff
@@ -0,0 +1,269 @@
+from django.db import models
+from django.contrib.postgres.fields import ArrayField
+from django.contrib.contenttypes.models import ContentType
+
+from enum import Enum
+from osf.utils.caching import cached_property
+
+
+def get_default_frequency_choices():
+    DEFAULT_FREQUENCY_CHOICES = ['none', 'instantly', 'daily', 'weekly', 'monthly']
+    return DEFAULT_FREQUENCY_CHOICES.copy()
+
+
+class NotificationType(models.Model):
+
+    class Type(str, Enum):
```

Comment:
@Johnetordoff clean-up the Type emum, maybe change syntax to `NotificationType.instance.NODE_TEST_TEST` more ergonomic.